### PR TITLE
Add queue job when submission created

### LINKED ISSUE #56: When creating a submission, queue it to be tested
After #55

### PR DESCRIPTION
**Pull request by matthewdu**
Resolves #56

### CODE REVIEW

File: spec/requests/submissions_spec.rb [hunk 1]
```diff
@@ -15,10 +16,6 @@
   let(:db_file) { create(:submission_db_file, name: uploaded_file.original_filename, contents: uploaded_file.read, content_type: 'text/plain') }
   let!(:submission) { create(:submission, user_id: student.id, problem_id: problem.id, db_file_id: db_file.id, language: FileHelper.filename_to_language(uploaded_file.original_filename)) }
 
-  # TODO(rwongone): Currently, we only check if a user is in a course's
-  # list of enrolled students to decide whether the resource is accessible.
-  # We will want to handle Submissions differently for teachers. How will
-  # a teacher submit a canon solution?
```
Comment (matthewdu) [inline spec/requests/submissions_spec.rb]: A canon solution submission is handled in `ProblemsController#create`

File: app/controllers/submissions_controller.rb [hunk 2]
```diff
@@ -20,22 +20,26 @@ def create
       return
     end
 
+    submission = Submission.new(
```
Comment (rwongone) [inline app/controllers/submissions_controller.rb]: What was the reason for moving this outside of the transaction?

Comment (matthewdu) [inline app/controllers/submissions_controller.rb]: so we can pass the submission variable to render

File: app/controllers/submissions_controller.rb [hunk 3]
```diff
@@ -20,22 +20,26 @@ def create
...
       )
 
-      render status: :created, json: submission
+      submission.db_file_id = file.id
+      submission.save!
     end
+
+    RunSubmissionsJob.perform_later(submission.id)
+
+    render status: :created, json: submission
```
Comment (jchorl) [inline app/controllers/submissions_controller.rb]: This is a bit weird. We'll have to be careful with error handling to ensure that if the transaction gets rolled back, the submission does not get returned.

Comment (matthewdu) [inline app/controllers/submissions_controller.rb]: Yeah, right now the transaction re-throws exceptions. `.save!` will throw an exceptions if it's not successful.

**Pull request merged**